woo-razorpay.php

```diff
@@ -248,14 +248,199 @@ public function getCustomOrdercreationMessage($thank_you_title, $order)
+            ];
+
+            $field = $this->get_field_key($key);
+
+            $defaults = array(
+                'title'       => '',
+                'type'        => 'multicheckbox',
+                'description' => '',
+                'desc_tip'    => false,
+                'options'     => array(),
+            );
+
+            $data = wp_parse_args( $data, $defaults );
+
+            ob_start();
+            ?>
+            <tr valign="top">
+                <th scope="row" class="titledesc">
+                    <label><?php echo wp_kses_post($data['title']); ?><?php echo $this->get_tooltip_html($data); ?></label>
+                </th>
+                <td class="forminp">
+                    <fieldset>
+                        <?php foreach ($data['options'] as $optionKey => $optionLabelDesc) : ?>
+                            <?php
+                                if ($field === $this->plugin_id . $this->id . '_' . 'enable_magic_checkout_group')
+                                {
+                                    $option_checked = $this->getSetting($c360FieldMapping[$optionKey]);
+                                }
+                                elseif (isset($c360FieldMapping[$optionKey]))
+                                {
+                                    $option_checked = get_option($c360FieldMapping[$optionKey]);
+                                }
+                            ?>
+                            <label for="<?php echo esc_attr($this->plugin_id . $this->id . '_' . $optionKey); ?>">
+                                <?php if (empty($optionLabelDesc['link']) === false) {?>
+                                    <a href="<?php echo esc_attr($optionLabelDesc['link']); ?>" target="_blank"><?php echo esc_html($optionLabelDesc['label']); ?></a>
+                                <?php } else { ?>
+                                    <input type="checkbox" name="<?php echo esc_attr($this->plugin_id . $this->id . '_' . $optionKey); ?>" id="<?php echo esc_attr($this->plugin_id . $this->id . '_' . $optionKey); ?>" value="1" <?php checked($option_checked, 'yes'); ?>>
+                                    <?php echo esc_html($optionLabelDesc['label']); ?>
+                                <?php } ?>
+                            </label><br>
+                            <p class="description"><?php echo esc_html($optionLabelDesc['description']); ?></p>
+                        <?php endforeach; ?>
+                    </fieldset>
+                </td>
+            </tr>
+            <?php
+            return ob_get_clean();
+        }
+
+        public function toggleRtbWidget($action ,$widgetEnabled = false)
+        {
+            $data = [
+                'rtb_widget_enabled' => $widgetEnabled,
+            ];
+
+            $rtbEnable = $this->rzpPostInstallationCall('save', $data);
+
+            // Experiment returned false
+            if (isset($rtbEnable['checkout360_status']) === true and
```

Comment:
@abdulwahidsharief since this api call(rzpPostInstallationCall) may return error in case of exception(return ['error' => $e->getMessage()]), can we put a check for the error first?

Reply:
@rohitcbr as error is logged in rzpPostInstallationCall just keeping a check for error to return will increase processing time for 200 response requests